src/components/draggable-list.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState,useRef } from 'react';
+import { Button, Tile } from 'carbon-components-react';
+import { css, cx } from 'emotion';
+import {
+	AddAlt32,
+	Draggable16,
+	TrashCan32
+} from '@carbon/icons-react';
+
+const addDragTargetStyle = css`
+	height: var(--drag-target-height, 32px);
+	width: 100%;
+	display: flex;
+	justify-content: center;
+	outline-offset: -2px;
+	transition: height 0.15s ease-in-out;
+`;
+
+const addButtonDraggingStyle = css`
+	outline: var(--outline, 1px dashed #1666fe);
+`;
+
+const addButtonStyle = css`
+	width: 100%;
+	display: flex;
+	justify-content: center;
+`;
+
+const trashButtonStyle = css`
+	margin-left: 12px;
+	top: 0;
+	right: 0;
+	border-color: transparent;
+
+	&.bx--btn.bx--btn--icon-only.bx--tooltip__trigger {
+		position: absolute;
+	}
+`;
+
+const draggableIconStyle = css`
+	position: absolute;
+	top: 50%;
+	left: 0;
+`;
+
+const tileStyle = css`
+	position: relative;
+`;
+
+export const DraggableTileList = ({
+	template, // Functional component
+	dragOver = (_: any) => { },	// Override onDragOver event
+	dataList,
+	setDataList,
+	defaultObject // Default object created
+}: any) => {
+	const [dragging, setDragging] = useState(false);
+	const draggedItem = useRef<HTMLDivElement>();
+	const [draggedIndex, setDraggedIndex] = useState(-1);
+
+	const onDragStart = (event: any, index: number) => {
+		setDragging(true);
+		setDraggedIndex(index);
+		event.dataTransfer.setData('item-data', JSON.stringify(dataList[index]));
+		draggedItem.current = event.currentTarget;
+	};
+
+	const onDragEnd = (_: any) => {
+		setDragging(false);
+		setDraggedIndex(-1);
+		draggedItem.current = undefined;
+	};
+
+	const onDragOver = (event: any) => {
+		// Prevent drop if user
+		if(dragOver(event) === false) {
+			return false;
+		}
+
+		event.preventDefault();
```

Comment:
Might need to
```suggestion
		event.preventDefault();

		if(dragOver(event) === false) {
			return false;
		}
```
🤔

Reply:
event.preventDefault should be AFTER the condition, otherwise the drop is allowed. By immediately returning false, the cursor css property is set to `no-drop` & prevents the the drop from making any changes.